scripts/target.rb

```diff
@@ -0,0 +1,42 @@
+module Build
+  class Target
+    TARGET_MAPPING = {
```

Comment:
Grouping the target mapping by the name may simplify things
```ruby
MAPPINGS = {
  'vsphere' => {
    'target' => 'vsphere',
    'extenstion' => 'ova'
  }
}

def initialize(name)
  @name = name = name.to_s
  @mapping = MAPPING[name]
end

def imagefactory_type
  @mapping['target']
end
```

Heck, or even:

```ruby
  class Target < OStruct(:name, :imagefactory_type, :file_extension)
  end

# this may not access the map 100%
def self.get_mapping(name)
  return Target.new(name, TARGET_MAPPING[name], FILE_EXTENSION[name])
end
```